lib/async.js

```diff
@@ -261,31 +274,28 @@ module.exports = function resolve(x, options, callback) {
     }
 
     function processDirs(cb, dirs) {
-        if (dirs.length === 0) return cb(null, undefined);
-        var dir = dirs[0];
+        iterate(0);
 
-        isDirectory(dir, isdir);
+        function iterate(i) {
+            if (i === dirs.length) return cb(null, undefined);
+            var dir = dirs[i];
 
-        function isdir(err, isdir) {
-            if (err) return cb(err);
-            if (!isdir) return processDirs(cb, dirs.slice(1));
-            var file = path.join(dir, x);
-            loadAsFile(file, opts.package, onfile);
-        }
+            loadAsFile(dir, opts.package, function (err, m, pkg) {
+                if (err) return cb(err);
+                if (m) return cb(null, m, pkg);
 
-        function onfile(err, m, pkg) {
-            if (err) return cb(err);
-            if (m) return cb(null, m, pkg);
-            loadAsDirectory(path.join(dir, x), opts.package, ondir);
-        }
+                loadAsDirectory(dir, opts.package, function (err, n, pkg) {
+                    if (err) return cb(err);
+                    if (n) return cb(null, n, pkg);
 
-        function ondir(err, n, pkg) {
-            if (err) return cb(err);
-            if (n) return cb(null, n, pkg);
-            processDirs(cb, dirs.slice(1));
+                    iterate(i + 1);
+                });
+            });
         }
     }
     function loadNodeModules(x, start, cb) {
-        processDirs(cb, nodeModulesPaths(start, opts, x));
+        processDirs(cb, packageIterator(x, start, function () {
```

Comment:
since the entire `processDirs` and `packageIterator` invocations both can't fit on one line, please format them so that each argument is on a line by itself.
```suggestion
        processDirs(
            cb,
            packageIterator(
              x,
              start,
              function () { return getPackageCandidates(x, start, opts); },
```